lib/models/Bootstrap.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+var async = require('async');
+var notif = require('./Notifcations');
+var virtualenv = require('./Virtualenv');
+var path = require('path');
+var fs = require('fs');
+var spinner = require('simple-spinner');
+var logger = require('log4js').getLogger('Bootstrap');
+
+var Bootstrap = function() {
+
+    /**
+     * Start bootstrap steps
+     * * *
+     * @param version - version number
+     * @param options - commander options
+     * @param fnCallback
+     */
+    function start(options, fnCallback) {
+        
+        async.waterfall([
+            
+            function(callback) {
+                notif.info('Initialize Cloudify...');
+                initialize(options, function(err){
+                    if(err) {
+                        notif.faild(err);
+                    }
+                    else {
+                        notif.ok('Initialization successful.');
+                        callback();
+                    }
+                });
+            },
+            
+            function(callback) {
+                var inputFile = options.prefix + '.' + options.inputs;
+                
+                notif.info('Bootstraping Cloudify...');
+                bootstrap(inputFile, options, function(err){
+                    if(err) {
+                        notif.faild(err);
+                    }
+                    else {
+                        notif.ok('Bootstrap Complete.');
+                        callback(null);
+                    }
+                });  
+            },
+
+            function(callback) {
+                getManagerIp(options, function(err, ip){
+                    var rootFolder = path.resolve('./');
+                    var ipFile = 'manager_ip.txt';
+                    
+                    notif.info('Writing manager ip address to ' + ipFile + '...');
+                    writeFile(ipFile, rootFolder, ip, function(err, file) {
+                        if(err) {
+                            notif.faild(err);
+                        }
+                        else {
+                            notif.ok(file + ' written success');
+                            callback(null, ip);
+                        }
+                    });
+                });
+            }
+            
+        ], fnCallback);
+        
+    }
+
+    /**
+     * Initializing Cloudify
+     * * *
+     * @param {object} options - commander options
+     * @param {function} callback
+     */
+    function initialize(options, callback) {
+        virtualenv.spawn('cfy init', options, callback);
+    }
+
+    /**
+     * Write files
+     * * * 
+     * @param {string} file - file name
+     * @param {string} folder - folder location
+     * @param {string} content - file content
+     * @param {function} callback
+     */
+    function writeFile(file, folder, content, callback) {
```

Comment:
Why do we need this function when we have this one? 
http://nodejs.org/api/fs.html#fs_fs_writefile_filename_data_options_callback